### ISSUE TITLE
Feature: packaged distributions.

### ISSUE THREAD

**Issue by robertdebock** [square/sudo_pair#21]
Hi,

Would it be an idea to package sudo-pair into shipable formats such as:
- binary in a tar.gz
- rpm
- deb
- apk

I'm able to assist in a RPM at least.

In order to have this feature, I think these items are required:
- A CI/CD pipeline to build packages if you release a new version of sudo-pair. Maybe travis-ci.
- (RPM) A spec file describing the build process and the installation.
- (deb) A control-file like the spec file.
- An artifact store.

I understand it's quite a request, so if I can help you anywhere, please involve me.

Regards, Robert de Bock.

**Comment by stouset** [comment]
I'm open to the plugin itself being packaged for distributions, but some care needs to be done when doing this. To enable it for real, you *really* need to configure it for your own environment: sudoer and pairer prompts, and `/etc/sudo.conf`.

**Comment by robertdebock** [comment]
Agreed, I guess you want to make it easy to install the plugin itself, to prevent people from needing to built it themselves.
You could include some sample configurations in /usr/share/doc or so.

**Comment by stouset** [comment]
I don't have time for that at the moment, but I'll leave this issue open since it's something I'd like to do. PRs welcome. :)

**Comment by robertdebock** [comment]
I'm working on the RPM build. Should start to work today or tomorrow. When it's working, I'll make a pull request.

**Comment by nbraud** [comment]
For what it's worth, I just filed an [Intent To Package bug](https://bugs.debian.org/900348) in Debian.

**Comment by stouset** [comment]
Thanks!

**Comment by stouset** [comment]
Did you ever get a chance to package this for Debian?

**Comment by RiotNrrrd** [comment]
Is this thing _sudo_pair_ or _sudo-pair_?  Please pick one or the other and stick with it, it's very confusing.

**Comment by stouset** [comment]
As far as I'm aware, every mention of the project is `sudo_pair`. If there's somewhere it's not, please let me know.

**Comment by RiotNrrrd** [comment]
> As far as I'm aware, every mention of the project is `sudo_pair`. If there's somewhere it's not, please let me know.

It's in _contrib/sudo-pair.spec_.

`[mockbuild@vm-rh7 SOURCES]$ tar -tzpf sudo_pair-v0.11.1.tar.gz | grep sudo-pair`
`sudo_pair-sudo_pair-v0.11.1/contrib/sudo-pair.spec`

`[mockbuild@vm-rh7 SOURCES]$ tar -xzpf sudo_pair-v0.11.1.tar.gz`

`[mockbuild@vm-rh7 SOURCES]$ find sudo_pair-sudo_pair-v0.11.1 -type f -print0 | xargs -0 egrep -l sudo-pair`
`sudo_pair-sudo_pair-v0.11.1/contrib/sudo-pair.spec`

`[mockbuild@vm-rh7 SOURCES]$ grep sudo-pair sudo_pair-sudo_pair-v0.11.1/contrib/sudo-pair.spec`
`Name: sudo-pair`

**Comment by robertdebock** [comment]
> > As far as I'm aware, every mention of the project is `sudo_pair`. If there's somewhere it's not, please let me know.
> 
> It's in _contrib/sudo-pair.spec_.
> 
> `[mockbuild@vm-rh7 SOURCES]$ tar -tzpf sudo_pair-v0.11.1.tar.gz | grep sudo-pair`
> `sudo_pair-sudo_pair-v0.11.1/contrib/sudo-pair.spec`
> 
> `[mockbuild@vm-rh7 SOURCES]$ tar -xzpf sudo_pair-v0.11.1.tar.gz`
> 
> `[mockbuild@vm-rh7 SOURCES]$ find sudo_pair-sudo_pair-v0.11.1 -type f -print0 | xargs -0 egrep -l sudo-pair`
> `sudo_pair-sudo_pair-v0.11.1/contrib/sudo-pair.spec`
> 
> `[mockbuild@vm-rh7 SOURCES]$ grep sudo-pair sudo_pair-sudo_pair-v0.11.1/contrib/sudo-pair.spec`
> `Name: sudo-pair`

That's something I can fix, I'll make a PR for it.
